pkg/controller/vitessbackupschedule/vitessbackupschedule_controller.go

```diff
@@ -466,76 +549,83 @@ func (r *ReconcileVitessBackupsSchedule) createJob(ctx context.Context, vbsc *pl
 		Spec: kbatch.JobSpec{
 			Template: corev1.PodTemplateSpec{
 				ObjectMeta: meta,
-				Spec:       pod,
+				Spec:       pod.Spec,
 			},
 		},
 	}
 
-	if err := ctrl.SetControllerReference(vbsc, job, r.scheme); err != nil {
+	if err := ctrl.SetControllerReference(&vbsc, job, r.scheme); err != nil {
 		return nil, err
 	}
 
-	return job, nil
-}
-
-func (r *ReconcileVitessBackupsSchedule) createJobPod(ctx context.Context, vbsc *planetscalev2.VitessBackupSchedule, name string) (pod corev1.PodSpec, err error) {
-	getVtctldServiceName := func(cluster string) (string, error) {
-		vtctldServiceName, vtctldServicePort, err := r.getVtctldServiceName(ctx, vbsc, cluster)
+	// Create the corresponding PVC for the new vtbackup pod
+	pvc := &corev1.PersistentVolumeClaim{}
+	key := client.ObjectKey{
+		Namespace: job.Namespace,
+		Name:      name,
+	}
+	err = r.client.Get(ctx, key, pvc)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return nil, err
+		}
```

Comment:
This could just be:
```
		if err != nil && !apierrors.IsNotFound(err) {
			return nil, err
		}
```

Then the rest of the code would be more clear and easier to read IMO. If we don't get an error, meaning that it already exists, we just re-use it? Do we ensure it's in a certain state in this case?

Reply:
good point, cases where the PVC already exist (thus no error) should be extremely rare as the name of the resource is salted and prefixed by the schedule UNIX time. but i have added an `else` condition to error in such situation: 072d109